spec/models/transformation_mapping_item_spec.rb

```diff
@@ -100,4 +106,82 @@
       end
     end
   end
+
+  # ---------------------------------------------------------------------------
+  # Network Validation
+  # ---------------------------------------------------------------------------
+  context "Network validation" do
+    let(:ems_vmware) { FactoryBot.create(:ems_vmware) }
+    let(:vmware_cluster) { FactoryBot.create(:ems_cluster, :ext_management_system => ems_vmware) }
+
+    let(:ems_redhat) { FactoryBot.create(:ems_redhat) }
+    let(:redhat_cluster) { FactoryBot.create(:ems_cluster, :ext_management_system => ems_redhat) }
+
+    let(:ems_ops) { FactoryBot.create(:ems_openstack) }
+    let(:cloud_tenant) { FactoryBot.create(:cloud_tenant_openstack, :ext_management_system => ems_ops) }
+
+    # source network
+    context "source vmware network" do
+      let(:src_vmware_host) { FactoryBot.create(:host_vmware, :ems_cluster => vmware_cluster) }
+      let(:src_switch) { FactoryBot.create(:switch, :host => src_vmware_host) }
+      let(:src_lan) { FactoryBot.create(:lan, :switch => src_switch) }
+
+      context "destination openstack" do
+        let(:dst_cloud_network) { FactoryBot.create(:cloud_network, :cloud_tenant => cloud_tenant) }
+
+        let(:tmi_ops_cluster) { FactoryBot.create(:transformation_mapping_item, :source => vmware_cluster, :destination => cloud_tenant) }
+        let(:ops_mapping) { FactoryBot.create(:transformation_mapping, :transformation_mapping_items => [tmi_ops_cluster]) }
+
+        let(:valid_source) { FactoryBot.create(:transformation_mapping_item, :source => src_lan, :destination => dst_cloud_network, :transformation_mapping_id => ops_mapping.id) }
+        let(:invalid_source) { FactoryBot.build(:transformation_mapping_item, :source => dst_cloud_network, :destination => src_lan, :transformation_mapping_id => ops_mapping.id) }
+
+        before do
+          src_switch.lans << [src_lan]
+          src_vmware_host.switches << [src_switch]
+          vmware_cluster.hosts << [src_vmware_host]
+
+          cloud_tenant.cloud_networks << [dst_cloud_network]
+        end
+
+        it "valid source" do
+          expect(valid_source.valid?).to be(true)
+        end
+        it "invalid source" do
+          expect(invalid_source.valid?).to be(false)
+        end
+      end
+
+      context "destination red hat" do
+        let(:dst_rh_host) { FactoryBot.create(:host_redhat, :ems_cluster => redhat_cluster) }
+        let(:dst_rh_switch) { FactoryBot.create(:switch, :host => dst_rh_host) }
```

Comment:
This has to be `:hosts => [dst_rh_host]`

Reply:
Done.